ci/ci/ci.py

```diff
@@ -253,7 +253,8 @@ def pr_requires_action(gh_username, pr_config):
     build_state = pr_config['build_state']
     review_state = pr_config['review_state']
     return (
-        is_pr_author(gh_username, pr_config) and (build_state == 'failure' or review_state == 'changes_requested')
+        is_pr_author(gh_username, pr_config)
+        and (build_state == 'failure' or review_state == 'changes_requested' or WIP in pr_config['labels'])
```

Comment:
What is the type of pr_config? I think it might be good to have types in the function signature

Reply:
I added a good deal of types, with a small bit of refactoring to help with the typing. The major addition here is the use of `TypedDict`, which allows us to enforce a fixed schema on a python dictionary. The conveniences and serialization of a dictionary with the type information we want from a class.